pathways/consignments.py

```diff
@@ -508,24 +514,37 @@ def _contaminated_boxes_to_cluster_sizes(contaminated_boxes, max_boxes_per_clust
         while sum_boxes < contaminated_boxes - max_boxes_per_cluster:
             sum_boxes += max_boxes_per_cluster
             cluster_sizes.append(max_boxes_per_cluster)
-        # add remaining boxes
+        # add last cluster with remaining contaminated boxes
         cluster_sizes.append(contaminated_boxes - sum_boxes)
         sum_boxes += contaminated_boxes - sum_boxes
         assert sum_boxes == contaminated_boxes
     else:
-        cluster_sizes = [contaminated_boxes]
+        cluster_sizes = [math.ceil(contaminated_boxes)]
     return cluster_sizes
 
 
-def create_stratas_for_clusters(num_units, cluster_width, cluster_sizes):
+def choose_strata_for_clusters(num_units, cluster_width, cluster_sizes):
     """Divide array of items or boxes into strata wide enough for clusters."""
```

Comment:
I've added more documentation for this function. Let me know if it still is not clear.

Reply:
This would mean touching a lot of places, but can `cluster_sizes: list of cluster sizes to contaminate` parameter just be number of clusters? It is always used with `len()`, right? I might be missing what _cluster sizes_ is in general though or a more general idea for the interface of this function.

Reply:
It is used to iterate through the lengths (often the last cluster is smaller than the rest) in the contamination step. But everywhere else, it's used to get the number of clusters. It could easily be a separate variable. That would make it easier to understand. Its a trade off with having another variable, but there are already a lot. Thoughts?

Reply:
Sorry, I was not clear, I meant just this function. By the other places, I meant calls of this function. I did not check if there are other cases like this.

Generally speaking, if both are need, then it depends, but calling `len()` is probably fine or even a better solution here given the high number of variable as you suggest.

I think what is confusing and can be fixed easily are places (i.e., functions) where only the number of clusters is needed. Passing list of cluster sizes there makes you wonder why these sizes are not used. Passing only the number of clusters makes it clear that using just that is indeed the intention. (You may still wonder why it is the case, but at least it is clear the code reflects the reasoning.)